src/fcntl.rs

```diff
@@ -381,3 +381,13 @@ pub fn fallocate(fd: RawFd, mode: FallocateFlags, offset: libc::off_t, len: libc
     let res = unsafe { libc::fallocate(fd, mode.bits(), offset, len) };
     Errno::result(res)
 }
+
+/// Allocate file space.
+///
+/// Ensure disk space is allocated for the file referred to by the file descriptor
+/// fd for the bytes in the range starting at offset and continuing for len bytes.
+#[cfg(any(target_os = "linux"))]
```

Comment:
This `#[cfg()]` statement is too narrow.  In addition to Linux, libc already defines `posix_fallocate` for FreeBSD, Dragonfly, and Android.  NetBSD has it too, though libc lacks the definition for NetBSD.  Please broaden the definition.  You will also need a matching `#[cfg()]` statement on the tests, or else nix's tests won't build on platforms that don't define this function.